### ISSUE TITLE
Remove all placeholder text and add text as permanent label

### ISSUE THREAD

**Issue by jeanniefu** [civiform/civiform#2617]
Feedback from a11y experts during an a11y audit.

Severity (Critical/Serious/Moderate/Good to have): Critical / good to have

Form fields should have persistent visible labels. Placeholder text that disappears is confusing. We should move the text to be a permanent label if needed.

Examples of all placeholder text:
Address:
<img width="1251" alt="Screen Shot 2022-06-01 at 4 40 47 PM" src="https://user-images.githubusercontent.com/9094240/171519782-eae180b2-76ee-4c7b-a593-a3ea90280422.png">
Enumerator:
<img width="1275" alt="Screen Shot 2022-06-01 at 4 46 34 PM" src="https://user-images.githubusercontent.com/9094240/171519828-4a587c9b-fd8a-410a-bef3-13b50960d6d7.png">
Date:
<img width="1249" alt="Screen Shot 2022-06-01 at 4 44 31 PM" src="https://user-images.githubusercontent.com/9094240/171519811-4729dd37-9caa-434f-ab1e-93fa614c7191.png">

In particular,
* **critical**: Enumerator placeholder text should be moved to be a permanent label for the individual field. (Programmatic label is the child name (which is sr-text only). Visually placeholder is used as a label. )
* **good to have**: Date placeholder (mm/dd/yyyy) should be a permanent label, on the review page and on the question page. This would help applicants validate whether their provided value matches the expected format.

Resources: 
* https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research

**Comment by jeanniefu** [comment]
My proposal is to update as following:

- Address:
  - Remove all placeholder text
  - Change first label from "Address line 1" to "Address line 1 (Street address)"
  - Change second label from "Unit/Apartment Number" to "Apartment, suite, unit, building, floor, etc."
- Date:
  - Question block page: copy the "mm/dd/yyyy" text as a 3rd line above the input
  - Application review page: add the "mm/dd/yyy" text below the question and above the answer:
<img width="1217" alt="Screen Shot 2022-06-09 at 12 05 25 PM" src="https://user-images.githubusercontent.com/9094240/172914784-8c6e4433-9c84-489b-a209-0a8525e4852c.png">
- Enumerator: See https://github.com/seattle-uat/civiform/issues/2624#issuecomment-1151432195

**Comment by rahaghas** [comment]
We generally should not need both a permanent label and placeholder text that has different content in it, so I agree with removing placeholder text in favor of putting all necessary context into the permanent labels. 

I would say (based a few inputs I've taken a look at online):

Address input:
line 1 label - "Address"
line 2 label - "Apartment, suite, etc. (optional)" -> in the event that the address input as a whole is required, we can clarify that this sub-input is not

For the date could we do something like:
label - "Date (mm/dd/yy)"

**Comment by jeanniefu** [comment]
Related date placeholder discussion: https://github.com/civiform/civiform/issues/3078. Tldr, the "mm/dd/yyyy" text raises some interesting considerations around localization/translation.
